spring-cloud-sleuth-zipkin/pom.xml

```diff
@@ -156,6 +156,16 @@
 			<artifactId>aspectjweaver</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>io.projectreactor</groupId>
+			<artifactId>reactor-core</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.reactivestreams</groupId>
+			<artifactId>reactive-streams</artifactId>
```

Comment:
We don't need this dependency. It is transitive from the `reactor-core`: https://github.com/reactor/reactor-core/blob/master/build.gradle#L247

The `SleuthAdvisorConfig` is not in `test`. Not sure how does it work since `reactor-core` is in the `test` scope 🤷‍♂️ .

Also you need to consider to have `reactor-core` as a `optional` dependency and move all the Reactor stuff to its own auto-config based on the `Flux` presence condition.

Does it make sense for you?

Reply:
> Also you need to consider to have reactor-core as a optional dependency and move all the Reactor stuff to its own auto-config based on the Flux presence condition.

@artembilan  Can you create new issue? It may take some time to refactor.

Reply:
There is one already: https://github.com/spring-cloud/spring-cloud-sleuth/issues/1074

Exactly that one drove me to take a look into your code.
So, you can consider to refactor and fix at the same time relying on that issue.

Reply:
Those issue should be fixed just by removing test scope. Don't see easy way how to make it optional again. Need to reach agreement in key concepts with contributors to refactor it. So new issue will be right place for this.

Reply:
I fixed the issue https://github.com/spring-cloud/spring-cloud-sleuth/commit/b13f15f9d7db49a0cda1e6777e01d0ad3deb61d8